ChangeLog3.d/remove_havege.txt

```diff
@@ -0,0 +1,3 @@
+API changes
+   * Remove HAVEGE module. HAVEGE is unsuitable for microcontrollers, therefore
+     it's impossible to guarantee it's safety.
```

Comment:
Thanks for adding an entry. However I think the justification should be expanded, as Mbed TLS is not only used on MCUs. Suggestion:
```Remove the HAVEGE module (previously disabled by default, used as a source of entropy when enabled). The design of HAVEGE makes in unsuitable for micro-controllers, and on platforms with a more complex CPU, usually the OS will provide better sources of entropy. Replacement: hardware RNG and/or seed file on micro-controllers, OS-provided entropy sources on other platforms.```

I think when possible, when we remove something we should suggest replacements for people who might have been using it. However, I'd be happy to hear from @gilles-peskine-arm or @yanesca about this last sentence: do you think this is appropriate advice in this case?

Reply:
I agree with the general principle of suggesting replacements. I don't think the opposition is exactly “micro-controllers” vs “OS”: many micros have an OS, which may or may not include drivers. Maybe something like

> The design of HAVEGE makes it unsuitable for micro-controllers, and platforms with a more complex CPU usually have a operating system interface that provides better randomness. Declare OS or hardware RNG interfaces with mbedtls_entropy_add_source() and/or use an entropy seed file created securely during device provisioning. See [knowledge base entry] for more information.

Reply:
data base link https://tls.mbed.org/kb/how-to/add-entropy-sources-to-entropy-pool
mentions HAVEGE module. Should data base be updated as well in scope of this task?

Reply:
We should update the KB, but as a separate task. (Especially since it's become cumbersome for us to update the existing website, so we aren't likely to make the new content available until we've migrated the documentation.)